input/psbt.go

```diff
@@ -0,0 +1,144 @@
+package input
+
+import (
+	"encoding/binary"
+
+	"github.com/btcsuite/btcd/btcec/v2"
+	"github.com/btcsuite/btcd/btcutil/psbt"
+	"github.com/lightningnetwork/lnd/keychain"
+)
+
+const (
+	// TODO(aakselrod): should this be encoded as a uint/varint instead?
+	// That would save space on the wire.
```

Comment:
Yes, I think those keys (`<keytype>` in the spec) are supposed to be "A [compact size](https://en.bitcoin.it/wiki/Protocol_documentation#Variable_length_integer) unsigned integer representing the type.". We have introduced lnd/btcwallet specific ones before, see https://github.com/lightningnetwork/lnd/blob/master/lnwallet/btcwallet/psbt.go#L26.

Reply:
I've done this, starting at `0x60` to not overlap with `0x51` for single tweaks, even though that's currently used per-input rather than per-output.